pkg/pillar/hypervisor/kvm.go

```diff
@@ -1079,38 +1257,15 @@ func (ctx KvmContext) CreateDomConfig(domainName string,
 			}
 		}
 	}
-	if len(pciAssignments) != 0 {
-		pciPTContext := struct {
-			PCIId        int
-			PciShortAddr string
-			Xvga         bool
-			Xopregion    bool
-		}{PCIId: netContext.PCIId, PciShortAddr: "", Xvga: false, Xopregion: false}
-
-		t, _ = template.New("qemuPciPT").Parse(qemuPciPassthruTemplate)
-		for _, pa := range pciAssignments {
-			short := types.PCILongToShort(pa.pciLong)
-			pciPTContext.Xvga = pa.isVGA()
-
-			if vendor, err := pa.vid(); err == nil {
-				// check for Intel vendor
-				if vendor == "0x8086" {
-					if pciPTContext.Xvga {
-						// we set opregion for Intel vga
-						// https://github.com/qemu/qemu/blob/stable-5.0/docs/igd-assign.txt#L91-L96
-						pciPTContext.Xopregion = true
-					}
-				}
-			}
 
-			pciPTContext.PciShortAddr = short
-			if err := t.Execute(file, pciPTContext); err != nil {
-				return logError("can't write PCI Passthrough to config file %s (%v)", file.Name(), err)
-			}
-			pciPTContext.Xvga = false
-			pciPTContext.Xopregion = false
-			pciPTContext.PCIId = pciPTContext.PCIId + 1
-		}
+	pciAssignmentsFiller := pciAssignmentsTemplateFiller{
+		multifunctionsDevices: multifunctionDevGroup(pciAssignments),
+		file:                  file,
+	}
+
+	err = pciAssignmentsFiller.do(file, pciAssignments, netContext.PCIId)
```

Comment:
The purpose of using `netContext.PCIId` in this function, which is unrelated to `netContext`, is simply to access the next `PICId` after the one used in `netContext`, right? This seems somewhat error-prone if someone restructures the code later...

Reply:
Yes, it is simply used for that.
Yes, it is error-prone, but hopefully if someone restructures the code later, they will run the tests.

Reply:
Do we have a test that can detect such an issue?

Reply:
Yes, because we do a string comparison of the whole config and actually I tried to refactor the code and ran into exactly that test failure: https://github.com/lf-edge/eve/commit/3f3c4f34227d130538a6216853246360e2b9f1ce